src/umbraco.cms/businesslogic/Packager/PackageInstance/CreatedPackage.cs

```diff
@@ -371,16 +413,14 @@ public void Publish()
             }
 
         }
-
+        
         private void AddDocumentType(DocumentType dt, ref List<DocumentType> dtl)
         {
-            if (dt.MasterContentType != 0)
+            if (dt.MasterContentType != 0 && dt.Parent.nodeObjectType == Constants.ObjectTypes.DocumentTypeGuid)

```

Comment:
Needed so that when this code is ran on a folder it doesn't crash. When it's not an actual document type the code in this `if` will be skipped.